client/landing/stepper/declarative-flow/internals/steps-repository/newsletter-setup/index.tsx

```diff
@@ -100,20 +66,7 @@ const NewsletterSetup: Step = ( { navigation } ) => {
 		setTagline( site.description );
 	}, [ site ] );
 
-	useEffect( () => {
-		if ( siteTitle.trim().length && invalidSiteTitle ) {
-			setInvalidSiteTitle( false );
-		}
-	}, [ siteTitle, invalidSiteTitle ] );
-
-	const imageFileToBase64 = ( file: Blob ) => {
-		const reader = new FileReader();
-		reader.readAsDataURL( file );
-		reader.onload = () => setBase64Image( reader.result as string );
-		reader.onerror = () => setBase64Image( null );
-	};
-
-	const onSubmit = async ( event: FormEvent ) => {
+	const handleSubmit = async ( event: FormEvent ) => {
```

Comment:
`NB`: Making note that you didn't create these changes @edanzer, but I'm curious why this needs to be an async function. It appears as if the `onSubmit` stepper method accepts synchronous functions, and, from what I can tell, there's no async action in the function body.

Reply:
@jeyip - Good point. And I'm not sure. My goal for the current PR was to try to keep everything as much the same as possible except what was needed to extract the shared form component. It does seem like this could be taken out here though.

Reply:
Could we address this in a follow-up? Or could we create a separate issue to track this?

Reply:
@jeyip - I'd be happy to handle this in an immediate follow up in the next day or two after PR is pushed.

Reply:
@jeyip - I went ahead and addressed this.